A55-xds-stateful-session-affinity.md

```diff
@@ -0,0 +1,516 @@
+  repeated HealthStatus override_host_status = 1;
+
+  repeated LoadBalancingConfig child_policy = 2;
+}
+```
+
+The `xds_cluster_resolver_experimental` config will be modified so that the
+`DiscoveryMechanism` inner message contains an additional
+`override_host_status` field of type `repeated HealthStatus`.
+
+`cds_experimental` policy will copy the
+[`common_lb_config.override_host_status`][or-host-status] value it gets
+in the CDS update from the `XdsClient` to the appropriate `DiscoveryMechanism`
+entry's `override_host_status` field in the config of the
+`xds_cluster_resolver_experimental` policy. If the field is unset,
+we treat it the same as if it was explicitly set to the default set which is
+[`UNKNOWN`, `HEALTHY`].
+
+The `xds_cluster_resolver_experimental` LB policy will copy the
+`override_host_status` value from its config's appropriate `DiscoveryMechanism`
+entry into the config for the `xds_override_host_experimental` policy. The
+diagram below shows how the configuration is passed down the hierarchy all the
+way from `cds_experimental` to `xds_override_host_experimental`.
+
+One of the existing policies (`xds_wrr_locality_experimental`,
+or `ring_hash_experimental`) is created as the child policy of
+`xds_override_host_experimental` as shown in the diagram below.
+
+![LB Policy Hierarchy Diagram](A55_graphics/lb-hierarchy.png)
+
+To propagate [`common_lb_config.override_host_status`][or-host-status] the
+following changes are required:
+
+* The [XdsClient][grpc-client-arch] will parse the
+  [common_lb_config.override_host_status][or-host-status] field from the
+  `Cluster` resource and copy the value to a new `override_host_status` field
+  in the CDS update that it sends to the watchers.
+
+* `cds_experimental` policy (which receives the CDS update) will copy the
+  value of `override_host_status` into the appropriate `DiscoveryMechanism`
+  entry's `override_host_status` field in the config of the
+  `xds_cluster_resolver_experimental` policy.
+
+* When the `xds_cluster_resolver_experimental` policy generates the config for
+  each priority, it will configure the `xds_override_host_experimental` policy
+  underneath the `xds_cluster_impl_experimental` policy. The generated config
+  for the `xds_override_host_experimental policy` will include the
+  `override_host_status` field from the `DiscoveryMechanism` associated with
+  that priority.
+
+### Notes
+
+Note that if stateful session affinity and retries (gRFC [A44][]) are both
+configured, all retry attempts will be routed to the same endpoint.
+
+### Temporary environment variable protection
+
+During initial development, this feature will be enabled by the
+`GRPC_EXPERIMENTAL_XDS_ENABLE_HOST_OVERRIDE` environment variable. This
+environment variable protection will be removed once the feature has
+proven stable. The environment variable protection will specifically apply
+to the following:
+
+* creation of `xds_override_host_experimental` LB policy
```

Comment:
I don't think we need to guard creation of this policy; the policy is a no-op if the HTTP filter is not created.

Reply:
good point. Done